tests/subsys/bluetooth/fast_pair/storage/src/main.c

```diff
@@ -0,0 +1,312 @@
+{
+	uint8_t *key_id = context;
+
+	return check_account_key_id(*key_id, account_key);
+}
+
+static void validate_unloaded(void)
+{
+	int err;
+	uint8_t all_keys[ACCOUNT_KEY_MAX_CNT][ACCOUNT_KEY_LEN];
+	size_t key_count = ACCOUNT_KEY_MAX_CNT;
+	uint8_t key_id = 0;
+
+	err = fp_storage_account_key_count();
+	zassert_equal(err, -ENODATA, "Expected error before settings load");
+
+	err = fp_storage_account_keys_get(all_keys, &key_count);
+	zassert_equal(err, -ENODATA, "Expected error before settings load");
+
+	err = fp_storage_account_key_find(all_keys[0], account_key_find_cb, &key_id);
+	zassert_equal(err, -ENODATA, "Expected error before settings load");
+}
+
+static void reload_keys_from_storage(void)
+{
+	fp_storage_clear_ram();
+	validate_unloaded();
+	int err = settings_load();
+
+	zassert_ok(err, "Failed to load settings");
+}
+
+static void setup_fn(void)
+{
+	validate_unloaded();
+
+	/* Load empty settings before the test to initialize fp_storage. */
+	int err = settings_load();
+
+	zassert_ok(err, "Settings load failed");
+}
+
+static void teardown_fn(void)
+{
+	fp_storage_clear_ram();
+	storage_mock_clear();
+	validate_unloaded();
+}
+
+static void test_unloaded(void)
+{
+	static const uint8_t test_key_id = 0;
+
+	int err;
+	uint8_t account_key[ACCOUNT_KEY_LEN];
+
+	validate_unloaded();
+	generate_account_key(test_key_id, account_key);
+
+	err = fp_storage_account_key_save(account_key);
+	zassert_equal(err, -ENODATA, "Expected error before settings load");
+}
+
+static void test_one_key(void)
+{
+	static const uint8_t test_key_id = 5;
+
+	int err;
+	uint8_t account_key[ACCOUNT_KEY_LEN];
+
+	generate_account_key(test_key_id, account_key);
+	err = fp_storage_account_key_save(account_key);
+	zassert_ok(err, "Unexpected error");
+
+	uint8_t read_keys[ACCOUNT_KEY_MAX_CNT][ACCOUNT_KEY_LEN];
+	size_t read_cnt = ACCOUNT_KEY_MAX_CNT;
+
+	zassert_equal(fp_storage_account_key_count(), 1, "Invalid Account Key count");
+	err = fp_storage_account_keys_get(read_keys, &read_cnt);
+	zassert_ok(err, "Getting Account Keys failed");
+	zassert_equal(read_cnt, 1, "Invalid Account Key count");
+	zassert_true(check_account_key_id(test_key_id, read_keys[0]), "Invalid key on read");
+
+	/* Reload keys from storage and validate them again. */
+	reload_keys_from_storage();
+	memset(read_keys, 0, sizeof(read_keys));
+	read_cnt = ACCOUNT_KEY_MAX_CNT;
+
+	zassert_equal(fp_storage_account_key_count(), 1, "Invalid Account Key count");
+	err = fp_storage_account_keys_get(read_keys, &read_cnt);
+	zassert_ok(err, "Getting Account Keys failed");
+	zassert_equal(read_cnt, 1, "Invalid Account Key count");
+	zassert_true(check_account_key_id(test_key_id, read_keys[0]), "Invalid key on read");
+}
+
+static void test_duplicate(void)
+{
+	static const uint8_t test_key_id = 3;
+
+	int err;
+	uint8_t account_key[ACCOUNT_KEY_LEN];
+
+	generate_account_key(test_key_id, account_key);
+	err = fp_storage_account_key_save(account_key);
+	zassert_ok(err, "Unexpected error");
```

Comment:
Please be more specific in the assert message e.g. Unexpected error during account key storage